client/me/purchases/purchases-list/index.jsx

```diff
@@ -44,6 +45,55 @@ class PurchasesList extends Component {
 		return ! this.props.sites.length && ! this.props.subscriptions.length;
 	}
 
+	renderPurchasesByOtherAdminsNotice() {
+		const { sites, translate } = this.props;
+
+		/*
+		 * Because this is only rendered when the user has no purchases,
+		 * we don't need to check each site to ensure the purchases aren't
+		 * linked with the user (they can't be, since they don't have any).
+		 */
+		const affectedSites = sites
+			.filter(
+				( site ) =>
+					( ! site.plan.is_free || site.products.length > 0 ) && site.capabilities.manage_options
+			)
+			.map( ( site ) => site.slug );
+
+		if ( ! affectedSites ) {
+			return;
+		}
+
+		let affectedSitesString = '';
+
+		if ( affectedSites.length === 1 ) {
+			affectedSitesString = affectedSites[ 0 ];
+		} else {
+			const allButLast = affectedSites.slice( 0, -1 ).join( ', ' );
+			const translatedAnd = translate( 'and', {
+				comment: 'last conjunction in a list of blognames: (blog1, blog2,) blog3 _and_ blog4',
+			} );
+			const last = affectedSites[ affectedSites.length - 1 ];
+
+			affectedSitesString = allButLast + ' ' + translatedAnd + ' ' + last;
```

Comment:
I do feel this is a bit clunky and fully open to recommendations - but looks like it's the approach used here (which is where I've also taken the string from to avoid the need for re-translation): https://github.com/Automattic/wp-calypso/blob/979e16217f0eb2874f21ae266021736f5f687741/client/reader/stream/x-post.jsx#L144-L160